nwb_conversion_tools/utils/basesinwbephyswriter.py

```diff
@@ -67,10 +67,8 @@ def add_waveforms(self):
             self.add_units_waveforms()
 
     def add_to_nwb(self, nwbfile: pynwb.NWBFile, metadata=None, **kwargs):
-        assert nwbfile is not None and isinstance(nwbfile, pynwb.NWBFile), "Instantiate an NWBFile and pass as argument"
-        self.metadata = metadata if metadata is not None else dict()
-        self.nwbfile = nwbfile
-        self._conversion_ops = kwargs
+        if self.nwbfile is None:
```

Comment:
I think you can avoid passing `nwbfile` here. We can use the `set_nwbfile` before calling this function!